User registration, login, token jwt, swagger, login register validation and MLu changes from previous PRs

GrzegorzRomanczyk opened the pull request.

GrzegorzRomanczyk committed:
add users and roles to database.. add swager.. refactor seeder

GrzegorzRomanczyk committed:
validacja usera

GrzegorzRomanczyk committed:
Generate token

GrzegorzRomanczyk committed:
a

GrzegorzRomanczyk committed:
token save in coocies.. admin view working only with admin role

michalLukaszewiczPrecast committed:
Created Transfer view

michalLukaszewiczPrecast committed:
Created initial transaction view

michalLukaszewiczPrecast committed:
Created NewTransactionDto, updated controlls on TransferView

michalLukaszewiczPrecast committed:
add Transfer Post method to controller

michalLukaszewiczPrecast committed:
Implemented Creating new transactions, need validation and testing, few problems on the frontend

michalLukaszewiczPrecast committed:
Updated TransactionService, added interface, cleaning

michalLukaszewiczPrecast committed:
Fixed Exception handling, Error displaying for user in processs

michalLukaszewiczPrecast committed:
Updated Seed.cs

michalLukaszewiczPrecast committed:
Fixed SuccessMessage

GrzegorzRomanczyk committed:
Merge pull request #4 from GrzegorzRomanczyk/MLu-add-new-transfer

Add Transfers

michalLukaszewiczPrecast committed:
Created test project, implemented Repository pattern for AccountDetails and Transactions

michalLukaszewiczPrecast committed:
implemented Repository Pattern, created RepositoryBase<T>

michalLukaszewiczPrecast committed:
Cleaning

michalLukaszewiczPrecast committed:
Implemented Validation in New Transaction form

michalLukaszewiczPrecast committed:
Added interfases, started writing TransactionServiceTests

michalLukaszewiczPrecast committed:
Writing TransactionServiceTests

michalLukaszewiczPrecast committed:
Created TransactionServiceTests

michalLukaszewiczPrecast committed:
Moved validation from AccountDetialRepository to TransactionService

michalLukaszewiczPrecast committed:
Removed interfaces from Models and Repositories
Fixed tests

michalLukaszewiczPrecast committed:
AccountDetailsRepositoryTests

GrzegorzRomanczyk committed:
Merge pull request #5 from GrzegorzRomanczyk/MLu-RepositoryPattern-Test-FixValidation

Repository pattern, testy

GrzegorzRomanczyk committed:
add login to menu, redirect when autorization failed

michLukaszewicz commented on MyBank/MyBankWebApp/Controllers/UserController.cs:
```diff
@@ -0,0 +1,93 @@
+        }
+
+        [HttpPost]
+        public IActionResult SendRegister(UserViewModel model)
```
Chińczyk z Udemy robił takie metody o tej samej nazwie (tutaj Register) i jedna była bez atrybutu (czytli HTTP GET)  a druga [HTTPPost]. Myślę że tak jest czytelniej ale jak masz jakieś argumenty przeciw to chętnie pogadam 😄

michLukaszewicz commented on MyBank/MyBankWebApp/Controllers/UserController.cs:
```diff
@@ -0,0 +1,93 @@
+                return View(model);
+            }
+
+            var dto = new RegisterUserDto
```
Ten ziomeczek też takie rzeczy nazywa ViewModel zamiast DTO - w mojej głowie to lepiej leży, ale nie do końca wiem do czego służy DTO

michLukaszewicz commented on MyBank/MyBankWebApp/Services/UserServices/UserService.cs:
```diff
@@ -0,0 +1,97 @@
+            };
+
+            var key = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(authenticationSettings.JwtKey));
+            var credentional = new SigningCredentials(key, SecurityAlgorithms.HmacSha256);
```
credential zamiast credentional?

michLukaszewicz commented on MyBank/MyBankWebApp/Views/User/Login.cshtml:
```diff
@@ -0,0 +1,17 @@
+﻿<html>
+<body>
+	<h1>Register new user</h1>
+	<form method="post" action="/User/SendLogin">
```
zamiast action użyjmy asp-action - możesz podejrzeć jak to się robi w [NewTransaction.cshtml] - wtedy asp sam generuję tą ścieżkę, którą masz tutaj ale dzieje się to bardziej automagicznie, dlatego jest to zalecana droga

michLukaszewicz commented on MyBank/MyBankWebApp/Views/User/Register.cshtml:
```diff
@@ -0,0 +1,35 @@
+﻿<html>
+<body>
+	<h1>Register new user</h1>
+	<form method="post" action="/User/SendRegister">
```
też asp-action zróbmy

GrzegorzRomanczyk replied:
ok

michLukaszewicz commented on MyBank/MyBankWebApp/Views/User/Login.cshtml:
```diff
@@ -0,0 +1,17 @@
+	<form method="post" action="/User/SendLogin">
+		<div asp-validation-summary="All" class="text-danger"></div>
+		<label for="email">Email:</label>
+		<input type="text" id="email" name="Email" required>
```
tych id="..." używasz do czegoś? jeśli nie to można je usunąć

GrzegorzRomanczyk replied:
nie mam pojecia. Mozliwe ze czat mi tak wyplul. Niech se sa :D

michalLukaszewiczPrecast committed:
Updated database, changed AccountDetails to Accounts, changed Id names, updated names (fixed typos)

michalLukaszewiczPrecast committed:
Update Database - changed typos, table names, types and maxLength of columns

michalLukaszewiczPrecast committed:
Created new tables for enums and updated Transactions

michalLukaszewiczPrecast committed:
FIx tests, cleaning

michalLukaszewiczPrecast committed:
Add AccountService ,
Add IAccountService to BankServiceController,
Update TransactionViewModel and Mapp,
Updated Transaction displaying

michalLukaszewiczPrecast committed:
Ordering History of transaction, styled it to have collors

michalLukaszewiczPrecast committed:
Cleaned BankServiceController, updated AccountService, enabled displaing the sender/reciver of transaction in transaction history

michalLukaszewiczPrecast committed:
Fixed top bar

michalLukaszewiczPrecast committed:
cleaning the frontend

GrzegorzRomanczyk committed:
Merge pull request #7 from GrzegorzRomanczyk/MLu-Update-database

Updated database

GrzegorzRomanczyk committed:
resolve comments
